### PR TITLE
Add Writers module

### LINKED ISSUE #22: Modules: IO
Export old code to support basic data saving to HDF5.

- [ ] MPI-aware HDF5 Writer
- [ ] XDMF description of the data

New features:

- [ ] Add HDF5 writer and XDMF description for Lagrangian markers, iso-surfaces and contours

### PR DESCRIPTION
**Pull request by luraess**
This PR will add IO capabilities relying on HDF5 and Xdmf. Closes #22 .

### CODE REVIEW

File: scripts_future_API/tm_stokes_mpi_wip.jl [hunk 1]
```diff
@@ -171,47 +172,50 @@ function main(; do_visu=false, do_save=false, do_h5_save=false)
...
+        Teff_min = 23 * 8 * prod(size(grid_l)) / ttot_max
+        Teff_max = 23 * 8 * prod(size(grid_l)) / ttot_min
+        printstyled("Performance: T_eff [min max] = $(round(Teff_min, sigdigits=4)) $(round(Teff_max, sigdigits=4)) \n"; bold=true, color=:green)
+    end
 
     if do_h5_save
         out_h5 = "results.h5"
-        @info "saving HDF5 file"
-        write_h5(joinpath(outdir, out_h5), fields, grid_g, comm, MPI.Info())
+        ndrange = CartesianIndices(((coordinates(topo)[1]*size(grid_l)[1]+1):(coordinates(topo)[1]+1)*size(grid_l)[1],
```
Comment (luraess) [inline scripts_future_API/tm_stokes_mpi_wip.jl:211]: @utkinis do you know if we have helpers for that?

File: scripts_future_API/tm_stokes_mpi_wip.jl [hunk 2]
```diff
@@ -59,7 +60,7 @@ function main(; do_visu=false, do_save=false, do_h5_save=false)
 
     grid_l = local_grid(grid_g, topo)
 
-    global_logger(FastIce.Logging.MPILogger(0, comm, global_logger()))
+    # global_logger(FastIce.Logging.MPILogger(0, comm, global_logger()))
```
Comment (luraess) [inline scripts_future_API/tm_stokes_mpi_wip.jl:63]: @utkinis this currently fails

File: src/Writers.jl [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+"""
+write_h5(path, fields, grid_l::CartesianGrid, grid_g::CartesianGrid, args...)
+
+Write `fields` in HDF5 format to a file in `path` using local grid `grid_l` and global grid `grid_g` information.
+"""
+function write_h5(path, fields, grid_l::CartesianGrid, grid_g::CartesianGrid, args...)
+    if !HDF5.has_parallel() && (length(args) > 0)
+        @warn("HDF5 has no parallel support.")
+    end
+    I = size(grid_l) == size(grid_g) ? CartesianIndices(size(grid_l)) : CartesianIndex( Int.(((origin(grid_l) .- origin(grid_g)) .÷ spacing(grid_l))) ) .+ CartesianIndices(size(grid_l))
```
Comment (luraess) [inline src/Writers.jl:20]: @utkinis this should now be dimension agnostic :-)

File: scripts_future_API/tm_stokes_mpi_wip.jl (hunk 2 above)
Comment (utkinis) [reply scripts_future_API/tm_stokes_mpi_wip.jl:63]: This could be addressed in a separate PR

**Pull request merged**